src/main/java/com/bobocode/service/impl/AdvisorServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.bobocode.service.impl;
+
+import static com.bobocode.enums.RoleType.ASSOCIATE;
+import static com.bobocode.enums.RoleType.PARTNER;
+import static com.bobocode.enums.RoleType.SENIOR;
+import static com.bobocode.enums.Status.NEW;
+import static java.math.BigDecimal.ZERO;
+import static java.math.BigDecimal.valueOf;
+
+import com.bobocode.domain.Advisor;
+import com.bobocode.domain.Application;
+import com.bobocode.domain.CreditRange;
+import com.bobocode.enums.RoleType;
+import com.bobocode.exception.NoAdvisorFoundException;
+import com.bobocode.exception.NoApplicationFoundException;
+import com.bobocode.repo.AdvisorRepository;
+import com.bobocode.repo.ApplicationRepository;
+import com.bobocode.service.AdvisorService;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AdvisorServiceImpl implements AdvisorService {
+
+  private final ApplicationRepository applicationRepository;
+  private final AdvisorRepository advisorRepository;
+
+  private final Map<RoleType, CreditRange> creditRangeByRoles = Map.of(
+      ASSOCIATE, new CreditRange(ZERO, valueOf(9999)),
+      PARTNER, new CreditRange(valueOf(10000), valueOf(50000)),
+      SENIOR, new CreditRange(valueOf(50001), valueOf(Integer.MAX_VALUE)));
+
+  @Override
+  @Transactional
+  public Application assignApplication(Long advisorId) {
+    log.info("Searching for advisor with id = {}", advisorId);
+    Advisor advisor = advisorRepository.findById(advisorId).orElseThrow(
+        () -> {
```

Comment:
This should be extracted into a helper method to avoid distraction from the main flow.